hr_contract_cae/models/hr_contract.py

```diff
@@ -18,21 +89,108 @@ class Contract(models.Model):
         track_visibility="onchange",
         help="Employee's monthly gross wage.",
         compute="_compute_wage",
+        copy=False,
     )
+    notes = fields.Text(copy=False)
+
+    @api.model
+    def create(self, vals):
+        # #  Todo: fix error when accessing self.count_type here: "Expected singleton". Maybe because the copy function is accessed both from current and initial contract?
+        # if self.count_type > self.type_id.max_usage:
+        #     raise ValidationError(
+        #         _(
+        #             "The maximum amount of %s contracts of type '%s' has been reached"
+        #             % (self.count_type, self.type_id)
+        #         )
+        #     )
+        res = super().create(vals)
+        if res.index_amendment == 0:
+            res.initial_contract_id = res.id
+        return res
+
+    @api.multi
+    @api.depends("employee_id", "initial_contract_id", "type_id")
+    def _compute_name(self):
+        for contract in self:
+            contract.name = "/".join(
+                filter(
+                    None,
+                    [
+                        (contract.employee_id.name or ""),
+                        (
+                            contract.initial_contract_id.type_id.name
+                            if contract.initial_contract_id.type_id.name
+                            and contract.index_amendment > 0
+                            else ""
+                        ),
+                        (contract.type_id.name or ""),
+                    ],
+                )
+            )
+
+    @api.multi
+    @api.depends("index_amendment")
+    def _compute_type_echelon(self):
+        for contract in self:
+            contract.type_echelon = (
+                "main" if contract.index_amendment == 0 else "amendment"
+            )
 
     @api.multi
     @api.depends("hours", "hourly_wage")
     def _compute_wage(self):
         for contract in self:
-            contract.wage = self.hours * self.hourly_wage
+            contract.wage = contract.hours * contract.hourly_wage
 
     @api.onchange("date_start", "duration")
     def onchange_date_start_duration(self):
         if self.date_start and self.duration:
-            self.date_end = self.date_start + relativedelta(months=self.duration)
+            self.date_end = self.date_start + relativedelta(
+                months=self.duration
+            )
 
     @api.onchange("date_end")
     def onchange_date_end(self):
         if self.date_start and self.date_end:
             rd = relativedelta(self.date_end, self.date_start)
             self.duration = rd.months + rd.years * 12
+
+    @api.multi
+    def create_amendment(self):
+        self.ensure_one()
+        latest_contract_id = self.latest_contract_id
+        contract = latest_contract_id.copy(
+            {
+                "state": "draft",
+                "type_id": self.env["hr.contract.type"]
```

Comment:
I'd leave it to False instead of embedding a sql request here.

Reply:
That was not possible since the copy() wants to create the object, and the field is required...